web/cypress/support/commands.ts

```diff
@@ -191,27 +189,107 @@ Cypress.Commands.add('checkRecordField', (field, name, values) => {
   });
 });
 
+
+Cypress.Commands.add('setupNetworkIdleTracking', (method: string = 'GET', urlPattern: string = '/api/**') => {
+  cy.wrap({
+    requestCount: 0,
+    lastRequestTime: 0,
+    startTime: Date.now()
+  }).as('networkIdleTracker');
+
+  cy.intercept(method, urlPattern, (req) => {
+    cy.get('@networkIdleTracker').then((tracker: Cypress.networkIdleTracker) => {
+      tracker.requestCount++;
+      tracker.lastRequestTime = Date.now();
+    });
+    req.continue();
+  }).as('networkIdleActivity');
+});
+
+
+Cypress.Commands.add('waitForNetworkIdle', (idleTime: number = 3000, timeout: number = 120000) => {
+  cy.get('@networkIdleTracker', { timeout: timeout })
+    .then((tracker: Cypress.networkIdleTracker) => {
+      const startTime = Date.now();
+
+      const checkIdleCondition = () => {
+        const now = Date.now();
+
+        if (tracker.requestCount > 0 && (now - tracker.lastRequestTime) >= idleTime) {
+          return true;
+        }
+
+        if (tracker.requestCount === 0 && (now - startTime) >= idleTime) {
```

Comment:
I don't understand this condition; if `requestCount == 0` , doesn't that mean no requests were intercepted and we should wait for intercept to happen in this case for /flow/metrics request?